src/coreclr/vm/interpexec.cpp

```diff
@@ -151,7 +152,7 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                     ip += 3;
                     break;
                 case INTOP_LDC_R8:
-                    LOCAL_VAR(ip[1], int64_t) = (int64_t)ip[2] + ((int64_t)ip[3] << 32);
```

Comment:
Hmm, actually think the actual issue was at this place. This isn't right. If you pass in e.g. ip[2] = 0x80000000 and ip[3] = 0x80000000, the result you'll get is 0x7fffffff80000000. The cast of ip[2] is signed, so (int64_t)ip[2] gets 0xffffffff80000000.

Reply:
That makes sense. Personally I think we should just use fixed-size memcpy in both places, since it's a float, instead of doing weird bit twiddling and two separate 32-bit operations. But maybe I'm missing something?